src/Functions/seriesOutliersDetectTukey.cpp

```diff
@@ -73,30 +70,26 @@ class FunctionSeriesOutliersDetectTukey : public IFunction
         if (input_rows_count == 0)
             return ColumnArray::create(ColumnFloat64::create());
 
-
         Float64 min_percentile = 0.25; /// default 25th percentile
         Float64 max_percentile = 0.75; /// default 75th percentile
         Float64 k = 1.50;
 
         if (arguments.size() > 1)
         {
-            Float64 p_min = arguments[1].column->getFloat64(0);
-            if (isnan(p_min) || !isFinite(p_min) || p_min < min_quartile|| p_min > max_quartile)
-                throw Exception(ErrorCodes::BAD_ARGUMENTS, "The second argument of function {} must be in range [2.0, 98.0]", getName());
+            static constexpr Float64 min_percentile_lower_bound = 0.02;
+            static constexpr Float64 max_percentile_upper_bound = 0.98;
 
-            min_percentile = p_min / 100;
+            min_percentile = arguments[1].column->getFloat64(0);
+            if (isnan(min_percentile) || !isFinite(min_percentile) || min_percentile < min_percentile_lower_bound|| min_percentile > max_percentile_upper_bound)
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "The second argument of function {} must be in range [2.0, 98.0]", getName());
 
-            Float64 p_max = arguments[2].column->getFloat64(0);
-            if (isnan(p_max) || !isFinite(p_max) || p_max < min_quartile || p_max > max_quartile || p_max < min_percentile * 100)
+            max_percentile = arguments[2].column->getFloat64(0);
+            if (isnan(max_percentile) || !isFinite(max_percentile) || max_percentile < min_percentile_lower_bound || max_percentile > max_percentile_upper_bound || max_percentile < min_percentile)
                 throw Exception(ErrorCodes::BAD_ARGUMENTS, "The third argument of function {} must be in range [2.0, 98.0]", getName());
 
-            max_percentile = p_max / 100;
-
-            auto k_val = arguments[3].column->getFloat64(0);
-            if (k_val < 0.0 || isnan(k_val) || !isFinite(k_val))
+            k = arguments[3].column->getFloat64(0);
+            if (k < 0.0 || isnan(k) || !isFinite(k))
                 throw Exception(ErrorCodes::BAD_ARGUMENTS, "The fourth argument of function {} must be a positive number", getName());
```

Comment:
In the docs it says `non-negative`, but here it says positive. The check allows `0`, so either change the error message; or change the docs and the check to prohibit `0` as `k`.